### DISCUSSION TITLE
Helm template error with nested label values in customConfig

### DISCUSSION THREAD

**Discussion by gabe-sorensen** [vectordotdev/helm-charts#417] [Q&A] [answered]
I am using the vector helm chart with the values file shown below, which works as expected.

```values.yaml
role: "Agent"
customConfig:
  data_dir: /vector-data-dir
  api:
    enabled: false
  sources:
    kubernetes_logs:
      type: kubernetes_logs
  sinks:
    loki:
      type: loki
      inputs: [kubernetes_logs]
      encoding:
        codec: json
      endpoint: http://loki-gateway
      labels:
        forwarder: "vector"
        cluster: "development"
        stream: "{{ .stream }}"
        source_type: "{{ .source_type }}"
        #k8s_pod_namespace: "{{ .kubernetes.pod_namespace }}"
```
However, when I un-comment `k8s_pod_namespace: "{{ .kubernetes.pod_namespace }}"` right below `source_type: "{{ .source_type }}"` I get the following error during a helm install:

```Error: template: vector/templates/daemonset.yaml:30:28: executing "vector/templates/daemonset.yaml" at <include (print $.Template.BasePath "/configmap.yaml") .>: error calling include: template: vector/templates/configmap.yaml:11:3: executing "vector/templates/configmap.yaml" at <tpl (toYaml .Values.customConfig) .>: error calling tpl: error during tpl function execution for "api:\n  enabled: false\ndata_dir: /vector-data-dir\nsinks:\n  loki:\n    encoding:\n      codec: json\n    endpoint: http://loki-gateway\n    inputs:\n    - kubernetes_logs\n    labels:\n      cluster: development\n      forwarder: vector\n      k8s_pod_namespace: '{{ .kubernetes.pod_namespace }}'\n      source_type: '{{ .source_type }}'\n      stream: '{{ .stream }}'\n    type: loki\nsources:\n  kubernetes_logs:\n    type: kubernetes_logs": template: vector/templates/daemonset.yaml:14:40: executing "vector/templates/daemonset.yaml" at <.kubernetes.pod_namespace>: nil pointer evaluating interface {}.pod_namespace```

I'm not sure why .source_type is working to extract the label, but .kubernetes.pod_namespace is complaining. Thanks!

**Comment by jszwedko** [comment, marked as answer]
Hi @gabe-sorensen !

The issue is that both Vector and Helm use `{{ }}` for templating. You'll need to escape it in your `values.yaml` to ensure that Helm doesn't interpret it as a template. You can find an example of this here:

https://github.com/vectordotdev/helm-charts/blob/develop/charts/vector/README.md#using-template-syntax-in-customconfig

**Reply by gabe-sorensen** [reply]
Ok that makes sense. I saw this discussion as well, and this seemed to fix my issue as well https://github.com/vectordotdev/helm-charts/issues/205 Thanks for the help!
